packages/text-bitmap/src/BitmapText.ts

```diff
@@ -241,12 +256,21 @@ export class BitmapText extends Container
                 pos.x += charData.kerning[prevCharCode];
             }
 
-            chars.push({
-                texture: charData.texture,
-                line,
-                charCode,
-                position: new Point(pos.x + charData.xOffset + (this._letterSpacing / 2), pos.y + charData.yOffset),
-            });
+            const charRenderData = charRenderDataPool.pop() || {
+                texture: Texture.EMPTY,
+                line: 0,
+                charCode: 0,
+                position: new Point(),
+            };
+
+            charRenderData.texture = charData.texture;
+            charRenderData.line = line;
+            charRenderData.charCode = charCode;
+            charRenderData.position.x = pos.x + charData.xOffset + (this._letterSpacing / 2);
+            charRenderData.position.y = pos.y + charData.yOffset;
+
+            chars.push(charRenderData);
```

Comment:
What do ya think about optimizing this one as well?

Instead of pushing each character to the (initially) zero-length array, do something like:

```js
// textLength is a _good_ heuristic of the actual no. of chars
const chars: CharRenderData = new Array(textLength);
let charsRendered = 0;

for (let i = 0; i < textLength; i++) {
     if (<skipCondition>) {
         continue;
     }  

     chars[charsRendered] = charREnderData;
     ++charsRenderered;
}

// Make sure undefined slots aren't at the end of the array, if needed
chars.length = charsRendered;
```

Reply:
@GoodBoyDigital I'm leaving this one on your will. It would be cool to do this optimization but not necessarily required.

Reply:
^^^